.github/workflows/unit_testing.yaml

```
@@ -0,0 +1,31 @@
+name: CMake
+
+on:
+  pull_request:
+    branches: [ "master" ]
+
+env:
+  # Customize the CMake build type here (Release, Debug, RelWithDebInfo, etc.)
+  BUILD_TYPE: Release
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Configure CMake
+      run: |
+        cmake \
+          -B ${{github.workspace}}/build \
+          -DCMAKE_BUILD_TYPE=${{env.BUILD_TYPE}} \
+          -DCMAKE_CXX_STANDARD=17 \
+          -DBUILD_TESTING=TRUE
```

```suggestion
      run: >
        cmake 
          -B ${{github.workspace}}/build 
          -DCMAKE_BUILD_TYPE=${{env.BUILD_TYPE}} 
          -DCMAKE_CXX_STANDARD=17 
          -DBUILD_TESTING=TRUE
```
IIRC you can use `>` instead of `|` to avoid the need for escapes.

---

Oh, I had no idea! That is nice. Thank you! Looking into it, you can't have extra indentation on proceeding lines in the command (like how I indented by two more spaces after the initial `cmake` line), so I'll make a commit based on your suggestion in a second.